modelci/hub/deployer/k8s/k8s_dispatcher.py

```diff
@@ -0,0 +1,149 @@
+import configparser
+from enum import Enum, unique
+
+from jinja2 import Environment, FileSystemLoader
+
+from modelci.types.bo import Engine
+from modelci.utils.misc import get_device
+
+jinja = Environment(
+    loader=FileSystemLoader('./k8s/templates'),
+    autoescape=True,
+    trim_blocks=True,
+    lstrip_blocks=True
+)
+
+@unique
+class StorageType(Enum):
+    """Enumerator of remote storage type.
+    """
+    NONE = 0
+    S3 = 1
+
+def generate_object_list(objects, uppercase_name: bool = False, is_env = False):
+    env_object_list = []
+    for name, value in objects.items():
+        env_object = {}
+        env_object['name'] = name.upper() if uppercase_name else name
+        if is_env and str(value).isdigit():
+            value = f'"{str(value)}"' # Digit in env must be with quotes. Due to a known issue of kubernetes: https://github.com/kubernetes/kubernetes/issues/82296
+        env_object['value'] = value
+        env_object_list.append(env_object)
+    return env_object_list
+
+def serve(
+        configuration: str = 'config.ini',
+        output_file_path: str = './k8s/deployment.yaml',
+        **additional_environment
+):
+    """Generate the kubernetes deployment yaml file with template.
```

Comment:
I think the serve function's behavior should be creating a k8s workload, currently this function is rendering the k8s config file. It's the pre step of `serve`, so we can rename it like `render`?

Reply:
I totally agree with you. `render` is a more proper name. To really do auto serving with k8s cluster, the project may need to integration with CI/CD tools to deploy the rendered file to k8s cluster. I will modify it.

Reply:
modified